CHANGES.md

```diff
@@ -110,6 +110,42 @@ a system action, its features will be added more in the future.
 
 ### Backward-incompatible API changes
 
+ -  Removed `DateTimeOffset?` type parameter that allowed a creation of
+    a genesis `Block<T>` with specific timestamp from
+    `BlockChain<T>.MakeGenesisBlock()`.  [[#2321]]
+ -  Overhauled constructors for `BlockMetadata`, `BlockContent<T>`,
+    `PreEvaluationBlockHeader`, `PreEvaluationBlock<T>`, `BlockHeader`,
+    and `Block<T>`.  [[#2321]]
+     -  All unsafe constructors have been removed in order to prevent
+        instantiation of invalid block related objects.
+     -  `BlockMetadata` has constructors `BlockMetadata(IBlockMetadata)`
+        and `BlockMetadata(long, DateTimeOffset, PublicKey, long, BigInteger,
+        BlockHash?, HashDigest<SHA256>)`.
+     -  `BlockContent` has constructors `BlockContent<T>(IBlockMetadata,
+        IEnumerable<Transaction<T>> transactions)`,
+        `BlockContent<T>(BlockMetadata)` and `BlockContent<T>(BlockMetadata,
+        IEnumerable<Trnasaction<T>> transactions)`.
+     -  `PreEvaluationBlockHeader` has constructors
+        `PreEvaluationBlockHeader(IPreEvaluationBlockHeader)` and
+        `PreEvaluationBlockHeader(BlockMetadata, (Nonce,
+        ImmutableArray<byte>))`.
+     -  `PreEvaluationBlock<T>` has constructors
+        `PreEvaluationBlock<T>(IPreEvaluationBlockHeader,
+        IEnumerable<Transaction<T>>)` and
+        `PreEvaluatoinBlock<T>(BlockContent<T>, (Nonce, ImmutableArray<byte>))`.
+     -  `BlockHeader` has constructors `BlockHeader(IBlockHeader)` and
+        `BlockHeader(PreEvaluationBlockHeader, (HashDigest<SHA256>,
+        ImmutableArray<byte>?, BlockHash))`.
+     -  `Block<T>` has constructors `Block<T>(IBlockHeader,
+        IEnumerable<Transaction<T>>)` and `Block<T>(PreEvaluationBlock<T>,
+        (HashDigest<SHA256>, ImmutableArray<byte>, BlockHash))`.
+ -  `BlockContent<T>` no longer inherits `BlockMetadata` and
+    `PreEvaluationBlock<T>` no longer inherits `PreEvaluationBlockHeader`.
+    [[#2231]]
+ -  Both `BlockMetadata` and `BlockContent<T>` are made immutable.  Their
+    properties can no longer be assigned to.  [[#2231]]
+ -  Copy extension methods for `BlockMetadata` and `BlockContent<T>` removed.
+    [[#2231]]
```

Comment:
```suggestion
 -  `BlockContent<T>` no longer inherits `BlockMetadata` and
    `PreEvaluationBlock<T>` no longer inherits `PreEvaluationBlockHeader`.
    [[#2231]]
 -  Both `BlockMetadata` and `BlockContent<T>` are made immutable.  Their
    properties can no longer be assigned to.  [[#2321]]
 -  Copy extension methods for `BlockMetadata` and `BlockContent<T>` removed.
    [[#2321]]
```

Reply:
Yup, but this is fixed in main release https://github.com/planetarium/libplanet/pull/2389.
It'll get fixed once we pull 0.43.0.

Reply:
Oh, Okay.